src/app/page/Signup/SignupScene.jsx

```diff
@@ -46,59 +45,64 @@ function SignupScene(props) {
     };
   }
 
-  function handleSignupSuccess(user) {
-    setValues({
-      ...values,
+  async function handleSignupSuccess(user) {
+    let data = {
+      ...getPayload(),
       phoneNumber: user.phoneNumber,
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
-    });
-    updateUser();
-    setActiveTab(Tabs.CONNECT);
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.CONNECT));
     return false;
   }
-  function handleConnectSuccess(user) {
-    setValues({
-      ...values,
+
+  async function handleConnectSuccess(user) {
+    let data = {
+      ...getPayload(),
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
-    });
-    updateUser();
-    setActiveTab(Tabs.PROFILE);
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.PROFILE));
     return false;
   }
-  async function updateUser() {
-    const payload = getPayload();
 
+  async function handleProfileSuccess() {
+    await updateUser({ ...getPayload() }, () => setActiveTab(Tabs.SUCCESS));
+  }
+
+  async function updateUser(payload, callback) {
+    payload = await getPayloadWithAddress(payload);
+    try {
+      User.update(payload.uid, payload).then(() => {
```

Comment:
I think we should stick with one pattern regarding async/await or promise chaining.

Couldnt this be: 

```
await User.update(payload.uid, payload)
```
?

Reply:
yeh lets use await here. and wrap all the logic with the try/catch

Reply:
I changed it to await, and removed the getPayloadWithAddress so it's all in the try/catch.